xc/device.ml

```diff
@@ -538,6 +547,12 @@ module Vbd_Common = struct
       | Disk -> "disk"
       | Floppy -> "floppy";
     ];
+    (* Hack: this should be returned separately from SMAPIv3 attach call *)
+    let (params, qemu_params) = if String.startswith "hack|" x.params then begin
+        match Stdext.Xstringext.String.split '|' x.params with
```

Comment:
You use Astring somewhere else. Please avoid using `Xstringext`

Reply:
Sounds good. This code was written by someone else; I'll update it.